doc/maintaining/cli.rst

```diff
@@ -0,0 +1,559 @@
+.. _cli:
+
+======================
+Command Line Interface (CLI)
+======================
+
+NB: From CKAN 2.9 onwards the CKAN configuration file is named 'ckan.ini'. Previous names: 'production.ini' and 'development.ini' (plus others) may also still appear in documentation and the software. These legacy names will eventually be phased out
+
+Most common CKAN administration tasks can be carried out from the command line
+on the server that CKAN is installed on, using the ``ckan`` command.
+
+If you have trouble running ckan commands, see
+`Troubleshooting ckan Commands`_ below.
+
+.. note::
+
+   Once you activate your CKAN virtualenv the "ckan" command is available from within any 
+   location within the host environment.
+
+   To run a ckan command without activating the virtualenv first, you have
+   to give the full path the ckan script within the virtualenv, for example:
+
+   .. parsed-literal::
+
+      |virtualenv|/bin/ckan -c |/path/to/ckan.ini| user list 
+
+   In the example commands below, we assume you're running the commands with
+   your virtualenv activated and from your ckan directory.
+
+The general form of a CKAN ``ckan`` command is:
+
+.. parsed-literal::
+
+   ckan --config=\ |/path/to/ckan.ini| **command** 
```

Comment:
I think you need to define `|/path/to/ckan.ini|` in this file: https://github.com/ckan/ckan/blob/master/doc/conf.py#L39 (I could be wrong!) or just do `/path/to/ckan.ini`.

And I think it might be easier for users if we give the default path to the file, as before, rather than `/path/to`. On both developer and server installs everyone just uses the default path, afaict.

Reply:
OK thanks David - Iv'e updated the doco to say "generally the location of the config file will be at /etc/ckan/default/ directory, however it can be located in _any_ directory on the host machine"

Reply:
That's a useful explainer. However I still think that in all other places in the docs, it should be `/etc/ckan/default/ckan.ini`, not `/path/to/ckan.ini`. So everyone can just copy and paste the command into their shell.

In addition, ckan.ini needs adding to conf.py, otherwise the pipe characters render (and they shouldn't):
<img width="321" alt="Screenshot 2020-02-07 at 11 51 18" src="https://user-images.githubusercontent.com/307612/74027705-94e74180-49a0-11ea-9ba0-10fdbe3a116e.png">

Reply:
And the conf.py problem is causing warnings on build:
```
/vagrant/src/ckan/doc/maintaining/getting-started.rst:81: WARNING: Undefined substitution referenced: "ckan.ini".
/vagrant/src/ckan/doc/maintaining/installing/install-from-package.rst:110: WARNING: Undefined substitution referenced: "ckan.ini".
/vagrant/src/ckan/doc/maintaining/installing/install-from-package.rst:120: WARNING: Undefined substitution referenced: "ckan.ini".
/vagrant/src/ckan/doc/maintaining/installing/install-from-source.rst:259: WARNING: Undefined substitution referenced: "ckan.ini".
/vagrant/src/ckan/doc/maintaining/installing/install-from-source.rst:450: WARNING: Undefined substitution referenced: "ckan.ini".
/vagrant/src/ckan/doc/maintaining/installing/install-from-source.rst:484: WARNING: Undefined substitution referenced: "ckan.ini".
```

Reply:
OK in the ../doc/conf.py file I have added this line: ".. |path/to/ckan.ini| replace:: |config_dir|/ckan.ini"